shared/net/addrs.go

```diff
@@ -78,23 +78,45 @@ func JoinAllHostPorts(addresses []string, ports []uint16) ([]string, error) {
 	return out, nil
 }
 
-// GetStringIPAddresses returns the list of IP addresses bound to the given
-// interfaces or all if none are given
+// GetStringIPAddresses returns a list of text IP addresses bound
+// to the given interfaces or all if none are given
 func GetStringIPAddresses(ifaces ...string) ([]string, error) {
+	addrs, err := GetNetIPAddresses(ifaces...)
+
+	out := make([]string, 0, len(addrs))
+	for _, addr := range addrs {
+		if s := addr.String(); s != "" {
+			out = append(out, s)
+		}
+	}
+
+	return out, err
+}
+
+// GetNetIPAddresses returns a list of net.IP addresses bound to
+// the given interfaces or all if none are given
+func GetNetIPAddresses(ifaces ...string) ([]net.IP, error) {
 	addrs, err := GetIPAddresses(ifaces...)
 
-	// even if GetIPAddresses() failed we convert whatever was returned
-	// before passing the error through
+	out := make([]net.IP, len(addrs))
+	for i, addr := range addrs {
+		var ip net.IP
+
+		if addr.Is4() || addr.Is4In6() {
+			a4 := addr.As4()
```

Comment:
I think this will panic on IPv6 mapped IPv4 addresses. 

"If ip is the zero Addr or an IPv6 address, As4 panics."

Reply:
using `addr.String()`
```
2023/01/28 23:49:05 interface[0/1]: "eth0"
2023/01/28 23:49:05 address: ::ffff:172.17.0.2
```
using `addr.As4()` for `adds.Is4() || addr.Is4in6()`
```
2023/01/29 19:43:35 interface[0/1]: "eth0"
2023/01/29 19:43:35 address: 172.17.0.2
```

but more than anything this is a reminder to make proper tests :-/

Reply:
No, wait, golang code says it will not panic in this case.
```go
	func (ip Addr) As4() (a4 [4]byte) {
	if ip.z == z4 || ip.Is4In6() {
		bePutUint32(a4[:], uint32(ip.addr.lo))
		return a4
	}
....
```